Documentation/SmartPointers.md

```diff
@@ -1,4 +1,4 @@
-# SerenityOS smart pointers
+# Smart pointers
```

Comment:
I think qualifying this as 'Ladybird smart pointers' or 'AK smart pointers' is a good idea, just to immediately clarify that we don't use smart pointers from the C++ standard library.